public/app/features/canvas/elements/icon.tsx

```diff
@@ -55,12 +46,7 @@ export function IconDisplay(props: CanvasElementProps) {
   };
 
   return (
-    <SanitizedSVG
-      onClick={onClick}
-      src={data.path}
-      style={svgStyle}
-      className={svgStyle.strokeWidth ? svgStrokePathClass : undefined}
-    />
+    <SanitizedSVG src={data.path} style={svgStyle} className={svgStyle.strokeWidth ? svgStrokePathClass : undefined} />
```

Comment:
Is there any reason to not support api editor functionality for the icon element as well? This has been a feature request that has come up [a few times](https://github.com/grafana/grafana/discussions/56835?sort=new#discussioncomment-3918128). We should probably disable the onclick call though when inline editing is enabled (similar to button)

Reply:
at very least we can create a follow-up issue to consider re-introducing support for this once we get button element in a good place

Reply:
The editor was commented out so the functionality was not exposed. Ryan also had some comments about it. I can create a follow up issue to bring it back if we decide there's value in that..